src/main/kotlin/io/kotlintest/Project.kt

```diff
@@ -0,0 +1,39 @@
+package io.kotlintest
+
+import org.reflections.Reflections
+
+object Project {
+
+  private var projectConfig: ProjectConfig? = null
+  private var executedBefore = false
+  private var executedAfter = false
+
+  init {
+    val configClasses = Reflections("io.kotlintest").getSubTypesOf(ProjectConfig::class.java)
+    if (configClasses.size > 1)  {
+      val configClassNames = configClasses.map { config -> config.simpleName }
+      throw InvalidConfigException("Duplicate GlobalConfig found: $configClassNames")
+    }
+    projectConfig = configClasses.firstOrNull()?.kotlin?.objectInstance
+  }
+
+  fun beforeAll() {
+    synchronized(executedBefore) {
```

Comment:
I think you meant `compareAndSet`? But since the result of the comparison is not useful here, I go with just `set(true)`.